lib/cooked_post_processor.rb

```diff
@@ -285,10 +285,11 @@ def create_span_node(klass, content=nil)
     span
   end
 
-  def update_topic_image
-    if @post.is_first_post?
-      img = extract_images_for_topic.first
-      @post.topic.update_column(:image_url, img["src"][0...255]) if img["src"].present?
+  def update_post_image
+    img = extract_images_for_post.first
```

Comment:
Shouldn't you iterate through all images until at least one as a valid "src" attribute?